src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -887,6 +881,35 @@ LogicalResult ONNXSeluOp::inferShapes(
 
 LogicalResult ONNXSequenceInsertOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
+  onnxmlir::SeqType seqType =
+      input_sequence().getType().dyn_cast<mlir::onnxmlir::SeqType>();
+  ShapedType tensorType = tensor().getType().dyn_cast<ShapedType>();
+  if (seqType.getLength() == 0) {
+    getResult().setType(onnxmlir::SeqType::get(tensorType, 1));
+    return success();
+  }
+
+  auto seqShape = seqType.getElementType().cast<ShapedType>().getShape();
+  auto seqRank = seqType.getElementType().cast<ShapedType>().getRank();
+  if (seqRank == -1)
+    return success();
+
+  auto tensorShape = tensorType.getShape();
+  auto tensorRank = tensorType.getRank();
+  if (tensorRank != seqRank)
+    return success();
```

Comment:
Can you just add one comment to explain this. For example
```
// Because shape inference propagation can only be performed on tensors of identical ranks, we return success here, leaving the output type as undefined.
```

I also wonder if you don't have to put an exception when you merge an "empty" list with a first insert... in that case, would you not directly take the first insert's shape?

Reply:
At line 887: direct take the insert's shape.